govcd/provider_vdc.go

```diff
@@ -182,3 +185,409 @@ func getProviderVdcByName(vcdClient *VCDClient, providerVdcName string, extended
 	}
 	return vcdClient.GetProviderVdcByHref(foundProviderVdcs.Results.VMWProviderVdcRecord[0].HREF)
 }
+
+// CreateProviderVdc creates a new provider VDC using the passed parameters
+func (vcdClient *VCDClient) CreateProviderVdc(params *types.ProviderVdcCreation) (*ProviderVdcExtended, error) {
+	if !vcdClient.Client.IsSysAdmin {
+		return nil, fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if params.Name == "" {
+		return nil, fmt.Errorf("a non-empty name is needed to create a provider VDC")
+	}
+	if len(params.ResourcePoolRefs.VimObjectRef) == 0 {
```

Comment:
Should we add a check if `ResourcePoolRefs`is not nil so it doesn't cause a `nil pointer dereference` panic in case someone doesn't pass the needed params?

Reply:
Yes. Thanks